### PR TITLE
Merge Constants Convention from Outlook Swift guide

### LINKED ISSUE #20: Integrate Outlooks' constants convention

### PR DESCRIPTION
**Pull request by cy255**
Fix #20 by adding additional Constants convention based on Outlook swift guide

-Added a section on Constants organization in FileContentOrdering.md

Note: the guidance in Outlook swift guide on putting constants on top of the file / class contradicts with our existing guidelines of ordering the file by access level, thus leaving that out of this PR.

### CODE REVIEW

File: FileContentOrdering.md [hunk 1]
```diff
@@ -62,3 +62,47 @@ internal func updateUnalliedHouses(_ houses: [House]) {
...
+    static let rotationAnimationspeed: CGFloat = 100
+    static let fadeAnimationdelay: CGFloat = 0.1
+    static let fadeAnimationduration: CGFloat = 0.25
+    static let fadeAnimationspeed: CGFloat = 200
+}
+
+class FruitSellerView: UIView {
+    ...
+
+    // good: split rotation and fade animation constants into different groups
```
Comment (markavitale) [inline FileContentOrdering.md:100]: Would it also make sense to have a bad example where we dump unrelated constants into a single struct? I would expect this to be bad pattern as well because we're not grouping according to content:

```swift
private struct Constants {
    static let rotationAnimationangle: CGFloat = .pi / 45.0
    static let rotationAnimationdelay: CGFloat = 0.15
    static let rotationAnimationduration: CGFloat = 0.3
    static let rotationAnimationspeed: CGFloat = 100
    static let fadeAnimationdelay: CGFloat = 0.1
    static let fadeAnimationduration: CGFloat = 0.25
    static let fadeAnimationspeed: CGFloat = 200
}
```

File: FileContentOrdering.md [hunk 2]
```diff
@@ -62,3 +62,47 @@ internal func updateUnalliedHouses(_ houses: [House]) {
...
+
+## Example
+
+```swift
+class FruitSellerView: UIView {
+    ...
+
+    // bad: no separation between rotation and fade animation constants, long constant names are harder to read
+
+    static let rotationAnimationangle: CGFloat = .pi / 45.0
```
Comment (markavitale) [inline FileContentOrdering.md]: nit: here and below, the capitalization is off. `rotationAnimationangle` should be `rotationAnimationAngle`

```suggestion
    static let rotationAnimationAngle: CGFloat = .pi / 45.0
```

File: FileContentOrdering.md [hunk 3]
```diff
@@ -62,3 +62,47 @@ internal func updateUnalliedHouses(_ houses: [House]) {
 private maximumHouseSize: Int = 50
 ```
 
+## Convention - Constants
+
+Use nested structs to group constants when possible.
```
Comment (markavitale) [inline FileContentOrdering.md]: A few questions here:
- We explicitly call out nested structs, but our example doesn't seem to use nested structs. Maybe just say we should use structs instead of saying nested structs?
- Should we be more specific and say the grouping should be logical, not just based on the structure of the constants?
    - I'd like to avoid `struct Constants {}` or `struct Strings {}` types of groupings that aren't actual logical groupings and more just dumping grounds for things.

**Review by markavitale — APPROVED** [review]

File: FileContentOrdering.md (hunk 3 above)
Comment (cy255) [reply FileContentOrdering.md]: I think the original intention of the "nested struct" is to demonstrate that the struct is a nested type within the class, but I can reword it to "Create nested structs within classes to group constants when possible; the grouping should be based on logical use instead of structure or type of the constants".

**Review by markavitale — APPROVED** [review]

**Review by anandrajeswaran — APPROVED** [review]
